### PR TITLE
Upgrade to webpack v4

### PR DESCRIPTION
**Pull request by lightnet328**
# Problem
Since UglifyJS included in the webpack currently used does not correspond to the syntax of ES6, the production build may fail.

# Solution

First of all, we need to use the new version of UglifyJS in order to prevent the production build from failing using ES6 syntax. And there are ways to use the new version of UglifyJS as follows.

1. Use an independent package
    - [webpack-contrib/uglifyjs-webpack-plugin](https://github.com/webpack-contrib/uglifyjs-webpack-plugin)
2. Upgrade to webpack v4
    - **This Pull Request proposes this method**
    - webpack v4 contains a new UglifyJS
3. Use an independent package and later use the packages contained in webpack

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -127,27 +125,29 @@
...
     "dev:server": "DEBUG=crowi:* NODE_ENV=development nodemon --watch ./lib/ --watch ./locales/ --ext 'js,json' --exec node app.js",
-    "dev:client": "NODE_ENV=development webpack --progress --profile --colors --display-error-details --watch --config tools/webpack.config.js",
+    "dev:client": "NODE_ENV=development webpack --progress --profile --colors --display-error-details --watch --mode=development --config tools/webpack.config.js",
     "flow": "flow",
     "format": "eslint --fix \"**/*.js\"",
     "lint": "eslint \"**/*.js\"",
     "test": "mocha -r test/bootstrap.js --globals chai --reporter dot test/**/*.test.js",
     "build": "npm run prod:client",
-    "prod:client": "NODE_ENV=production webpack --bail --progress --colors --display-error-details --config tools/webpack.config.js",
+    "prod:client": "NODE_ENV=production webpack --bail --progress --colors --display-error-details --mode=production --config tools/webpack.config.js",
```
Comment (sotarok) [inline package.json]: I removed `--progress` on #313 :)

File: package.json [hunk 2]
```diff
@@ -127,27 +125,29 @@
...
+    "sinon-chai": "~2.12.0",
+    "webpack-cli": "^3.0.8"
   },
   "license": "MIT",
   "scripts": {
     "start": "node app.js",
     "dev": "concurrently --kill-others \"npm run dev:server\" \"npm run dev:client\" \"npm run watch\" ",
     "dev:server": "DEBUG=crowi:* NODE_ENV=development nodemon --watch ./lib/ --watch ./locales/ --ext 'js,json' --exec node app.js",
-    "dev:client": "NODE_ENV=development webpack --progress --profile --colors --display-error-details --watch --config tools/webpack.config.js",
+    "dev:client": "NODE_ENV=development webpack --progress --profile --colors --display-error-details --watch --mode=development --config tools/webpack.config.js",
```
Comment (dyong0) [inline package.json]: DRY, mode is the same as `NODE_ENV`
```js
const env = process.env.NODE_ENV

// webpack config
{
    mode: env
}
```

Comment (lightnet328) [reply package.json]: Surely. I replaced it with environment variables.

File: package.json (hunk 1 above)
Comment (lightnet328) [reply package.json]: Thanks for letting me know.
Since `mode` option refers to an environment variable, merge conflict does not occur now.

**Pull request merged**